src/Constants/authorization.js

```diff
@@ -0,0 +1,118 @@
+// Spotify application details
+const stateKey = 'spotify_auth_state';
+const CLIENT_ID = 'c73c8dd43ae64b7c82a1e3b355cda443';
+const scope = 'playlist-modify-private playlist-modify-public user-read-private user-read-email';
+
+function getRedirectUri() {
+    const origin = window.origin;
+    if (origin.includes('localhost')) {
+        return 'http://localhost:8080';
+    }
+    return origin;
+}
+
+// Function to generate a random string
+function generateRandomString(length) {
+    let text = '';
+    let possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+
+    for (let i = 0; i < length; i++) {
+        text += possible.charAt(Math.floor(Math.random() * possible.length));
+    }
+    return text;
+};
+
+// Function to set the state in localStorage
+function setStateInStorage(stateKey) {
+    const state = generateRandomString(16);
+    localStorage.setItem(stateKey, state);
+};
+
+// Function to handle login
+const handleLogin = (e) => {    
+    e.preventDefault();
+
+    //Set the state in localStorage
+    setStateInStorage(stateKey);
+
+    // Get the generated state from localStorage
+    const storedState = localStorage.getItem(stateKey);
+
+    // Build the authorization URL using the generated state
+    let url = 'https://accounts.spotify.com/authorize';
```

Comment:
There is a URL class in javascript that you can take advantage of here

Reply:
Done